api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalMaxPool1DTest.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.core.layer
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.pooling.GlobalMaxPool1D
+import org.jetbrains.kotlinx.dl.api.core.shape.shapeToIntArray
+import org.jetbrains.kotlinx.dl.api.inference.keras.CHANNELS_FIRST
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.tensorflow.EagerSession
+import org.tensorflow.Graph
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+
+internal class GlobalMaxPool1DTest {
+    @Test
+    fun default() {
+        val input = arrayOf(
+            arrayOf(
+                floatArrayOf(1.0f, -2.0f, 3.0f),
+                floatArrayOf(0.5f, 2.0f, 5.0f)
+            ),
+            arrayOf(
+                floatArrayOf(5.0f, 3.0f, 1.0f),
+                floatArrayOf(6.0f, -2.5f, 4.0f),
+            )
+        )
+        val expected = arrayOf(
+            floatArrayOf(1.0f, 2.0f, 5.0f),
+            floatArrayOf(6.0f, 3.0f, 4.0f)
+        )
+        val layer = GlobalMaxPool1D()
+
+        EagerSession.create().use {
+            val tf = Ops.create()
+            val inputShape = Shape.make(input.size.toLong(), input[0].size.toLong(), input[0][0].size.toLong())
+            layer.build(tf, KGraph(Graph().toGraphDef()), inputShape)
+
+            val inputOp = tf.constant(input)
+            val isTraining = tf.constant(true)
+            val numberOfLosses =  tf.constant(1.0f)
+            val output = layer.forward(tf, inputOp, isTraining, numberOfLosses).asOutput()
+
+            // Check output shape is correct.
+            val expectedShape = intArrayOf(input.size, input[0][0].size)
+            Assertions.assertArrayEquals(
+                expectedShape,
+                shapeToIntArray(output.shape())
+            )
+
+            // Check output values are correct.
+            val actual = Array(input.size) { FloatArray(input[0][0].size) }
+            output.tensor().copyTo(actual)
+            for (i in expected.indices) {
+                Assertions.assertArrayEquals(
+                    expected[i],
+                    actual[i]
+                )
+            }
+        }
+    }
+
+    @Test
+    fun withDataFormat() {
+        val input = arrayOf(
+            arrayOf(
+                floatArrayOf(1.0f, -2.0f, 3.0f),
+                floatArrayOf(0.5f, 2.0f, 5.0f)
+            ),
+            arrayOf(
+                floatArrayOf(5.0f, 3.0f, 1.0f),
+                floatArrayOf(6.0f, -2.5f, 4.0f),
+            )
+        )
+        val expected = arrayOf(
+            floatArrayOf(3.0f, 5.0f),
+            floatArrayOf(5.0f, 6.0f)
+        )
+        val layer = GlobalMaxPool1D(dataFormat = CHANNELS_FIRST)
```

Comment:
remove this test due to removed dataFormat

Reply:
Done!